modules/templates/helper.go

```diff
@@ -237,8 +237,8 @@ func DotEscape(raw string) string {
 
 // Iif is an "inline-if", similar util.Iif[T] but templates need the non-generic version,
 // and it could be simply used as "{{Iif expr trueVal}}" (omit the falseVal).
-func Iif(condition bool, vals ...any) any {
-	if condition {
+func Iif(condition any, vals ...any) any {
+	if condition != nil && condition.(bool) {
```

Comment:
I think `Iif(condition bool, vals ...any) any` is right.

Or, you need something like what `IfZero` does, to follow the golang template `if` behavior.

Reply:
I agree but when use the function in go template, sometimes it will give a `nil` if the variable doesn't exist. i.e. `{{Iif .IsActive ...}}` . If the map/struct have no key/field named `IsActive` it will return `nil`. If put another `IsZero` here will make it less readable.

Or we can have another method.

Reply:
> Or we can have another method.

No "another" please.

Reply:
```go
var zero any
fmt.Printf("%v is nil: %v", zero, zero == nil)

//output:
<nil> is nil: true
```  
When the type is declared as `any`, the `zero value` is `nil`. Therefore, it should suffice to simply verify that it is not `nil`, shouldn't it?

Reply:
`0`, `""` are all zero-value in Golang

Reply:
https://pkg.go.dev/text/template

![image](https://github.com/go-gitea/gitea/assets/2114189/e8daaff2-74d5-43ee-91ed-e543dd921af5)

Reply:
It needs an `IsEmpty` helper to match Golang template `if` behavior:

```
func IsEmpty[T comparable](v T) bool {
    var zero T
    if v == zero { return true}
    if slice { return len == 0}
    if map ( return len == 0}
    return false
}
```

Reply:
Please take a look at my latest modifications to see if they still need any adjustments.